bika/lims/exportimport/instruments/resultsimport.py

```diff
@@ -834,6 +872,8 @@ def _process_analysis(self, objid, analysis, values):
 
         if resultsaved or len(interimsout) > 0:
             doActionFor(analysis, 'submit')
+            self.calculateTotalResults(objid, analysis)
```

Comment:
You are missing to trigger the calculation for the current analysis. `self.calculateTotalResults(objid, analysis)` only performs the calculation of results for the analyses that depends on the current analysis, but not for the analysis itself. One option would be to restore the L810. Another option would be to add `analysis.calculateResult(override=self._override[1])` just after `self.calculateTotalResults(objid, analysis)`

TL;DR
There are Calculations that do not require of results from other Analyses and only use Interim fields. Results for analyses from this type will not be calculated. In fact, the calculation of results for analyses with only interims was correctly performed before your change, but calculation of results for analyses that depend on other analyses were not calculated. Now, happens the other way round.